### PR TITLE
fix: external module import order

### PR DESCRIPTION
**Pull request by underfin**
### Description

The pr fix external module import order(`import_external` test).

#### Module Ast Part
The pr implement `Ast Part`, a part is composed of some ast stmts. 

``` .ts
// main.js
import 'external'  // execute first
import 'a.js'
```
See the above module the external module should execute first, but before implement will render to the following code.

``` .ts
// a.js
xxx
// main.js
import 'external'
```

The part is used to generate correct code execute order, see the following.

``` .ts
// main.js
import 'external'
// a.js
xxx
```

The `main.js` has `external` part and `a.js` part, it will scanned ast first scan ast.

#### Chunk Part Codegen

Before chunk codegen is implemented by module ast codegen, the pr will collect chunk parts by visite chunk module part. The chunk render is also used ast part to gen some ast stmts. 

### Test Plan

Update snapshot.

---

### CODE REVIEW

File: crates/rolldown/src/bundler/graph/graph.rs [hunk 1]
```diff
@@ -57,6 +57,8 @@ impl Graph {
       match action {
         Action::Enter => {
           if !entered_ids.contains(&id) {
+            *module.exec_order_mut() = next_exec_order;
+            next_exec_order += 1;
```
Comment (underfin) [inline crates/rolldown/src/bundler/graph/graph.rs:61]: Before only keep execute module order, but shouln't keep module execute stmts order.

File: crates/rolldown/src/bundler/visitors/scanner.rs [hunk 2]
```diff
@@ -11,14 +11,15 @@ use oxc::{
...
 };
 use rolldown_oxc::BindingIdentifierExt;
 use rustc_hash::FxHashMap;
 
 #[derive(Debug, Default)]
 pub struct ScanResult {
+  pub parts: IndexVec<PartId, Part>,
   pub named_imports: FxHashMap<SymbolId, NamedImport>,
   pub named_exports: FxHashMap<Atom, LocalOrReExport>,
   pub stmt_infos: IndexVec<StmtInfoId, StmtInfo>,
```
Comment (hyf0) [inline crates/rolldown/src/bundler/visitors/scanner.rs:25]: We already have the similar struct like `Part`, which is called `StmtInfo`.

File: crates/rolldown/tests/esbuild/default/export_forms_es6/artifacts.snap [hunk 3]
```diff
@@ -27,5 +16,16 @@ function Fn () {
 class Class {
 }
 
+// a.js
```
Comment (hyf0) [inline crates/rolldown/tests/esbuild/default/export_forms_es6/artifacts.snap:19]: The old snapshot show a wrong order, which is a bug, but the new snapshot seems also wrong about the order between `a.js` and `main.js`.

**Comment by hyf0** [comment]
I currently don't prefer to use `Part` or `StmtInfo` to solve the order problem, it adds unnecessary complexity for solving the problem.

File: crates/rolldown/src/bundler/graph/graph.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/src/bundler/graph/graph.rs:61]: I don't get your words.

**Pull request closed without merging**